### ISSUE TITLE
Question on documentation

### ISSUE THREAD

**Issue by wildcolor** [TA-Lib/ta-lib-python#296]
I am very new to ta-lib. 

Here is a very stupid question about the documentation. Can anyone direct me **where exactly** is the **detailed** documentation? For example, the following shows BBANDS has MA_Type.T3, which means BollingerBands will use EMA instead of SMA. But I can't find the description about BBANDS and MA_Type anywhere..... 

    upper, middle, lower = talib.BBANDS(close, matype=MA_Type.T3)

**Comment by mrjbq7** [comment]
The best documentation is to look at the underlying C library source code:

http://www.ta-lib.org/

This project is just a lightweight wrapper to allow calling the C library from Python.

**Comment by wildcolor** [comment]
Fair enough to look at the C source. But no detailed documentation, API specification etc? Have to look at the source code?

> The best documentation is to look at the underlying C library source code: http://www.ta-lib.org/ This project is just a lightweight wrapper to allow calling the C library from Python.
> […](#)
> On Jan 26, 2020, at 9:36 PM, Wei ***@***.***> wrote: ﻿ I am very new to ta-lib. Here is a very stupid question about the documentation. Can anyone direct me where exactly is the detailed documentation? For example, the following shows BBANDS has MA_Type.T3, which means BollingerBands will use EMA instead of SMA. But I can't find the description about BBANDS and MA_Type anywhere..... upper, middle, lower = talib.BBANDS(close, matype=MA_Type.T3) — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by mrjbq7** [comment]
You can search for documentation for the Ta-Lib C library, that is applicable. I have tried not to have too much documentation here since this is just a lightweight wrapper. 

Maybe that makes it difficult sometimes, not sure.

**Comment by wildcolor** [comment]
Thanks for the reply. But I still can't find a decent official description about the talib functions, eg API specification for BBANDS?

Being one of the most well known library, I feel really puzzled to not to see official documentation...

> You can search for documentation for the Ta-Lib C library, that is applicable. I have tried not to have too much documentation here since this is just a lightweight wrapper. Maybe that makes it difficult sometimes, not sure.
> […](#)
> On Jan 27, 2020, at 6:58 AM, Wei ***@***.***> wrote: ﻿ Fair enough to look at the C source. But no detailed documentation, API specification etc? Have to look at the source code? The best documentation is to look at the underlying C library source code: http://www.ta-lib.org/ This project is just a lightweight wrapper to allow calling the C library from Python. … On Jan 26, 2020, at 9:36 PM, Wei @.***> wrote: ﻿ I am very new to ta-lib. Here is a very stupid question about the documentation. Can anyone direct me where exactly is the detailed documentation? For example, the following shows BBANDS has MA_Type.T3, which means BollingerBands will use EMA instead of SMA. But I can't find the description about BBANDS and MA_Type anywhere..... upper, middle, lower = talib.BBANDS(close, matype=MA_Type.T3) — You are receiving this because you are subscribed to this thread. Reply to this email directly, view it on GitHub, or unsubscribe. — You are receiving this because you commented. Reply to this email directly, view it on GitHub, or unsubscribe.

**Comment by mrjbq7** [comment]
Maybe you’ll have some benefit from looking at these descriptions and links:

http://tadoc.org/

**Comment by claytonEtillman** [comment]
This is where I found a reference to the available MA_Type:

https://fja05680.github.io/TA-Lib-tutorial/

> Print the available moving average types
> 
> `attributes = [attr for attr in dir(MA_Type)` 
> `              if not attr.startswith('__')]`
> `attributes`
> $ ['DEMA', 'EMA', 'KAMA', 'MAMA', 'SMA', 'T3', 'TEMA', 'TRIMA', 'WMA', '_lookup']
>
> `MA_Type.__dict__`
> $ {'_lookup': {0: 'Simple Moving Average',
> $  1: 'Exponential Moving Average',
> $  2: 'Weighted Moving Average',
> $  3: 'Double Exponential Moving Average',
> $  4: 'Triple Exponential Moving Average',
> $  5: 'Triangular Moving Average',
> $  6: 'Kaufman Adaptive Moving Average',
> $  7: 'MESA Adaptive Moving Average',
> $  8: 'Triple Generalized Double Exponential Moving Average'}}
>
> `print(MA_Type[MA_Type.DEMA])`
> $ Double Exponential Moving Average

**Comment by claytonEtillman** [comment]
I also found this site which has a little more info on the functions:

https://ta-lib.github.io/ta-doc/index.htm
